### PR TITLE
Add Gondo's Upgrades as Progressive Items

### PR DESCRIPTION
**Pull request by RaidWhipple**
Made the Bow, Slingshot, and Bug Net progressive and added their respective Scrap Shop upgrades as well as the Beetles.

Added a macro for the Swift Beetle and applied it to Secret Passage access and the Waterfall Island Crystal.

Gondo is still available, but there are no as of yet discovered issues with him.  He properly responds to your current inventory, and finding a progressive item for an item that is already fully upgraded does nothing.

Added an option to the UI for these progressive items.  When off, 2 Progressive Beetles, 2 Progressive Bows, 1 Progressive Slingshot, and 1 Progressive Bug Net are removed from all_progressive_items before the unplaced lists are created.  I also rearranged the UI a bit to make room for it by adding an Item Pool section to Additional Settings.

### CODE REVIEW

**Review by Battlecats59 — CHANGES_REQUESTED** [review]
In `checks.yaml`, you need to set the `original item` for the three following checks to be progressive. Especially for the bug net, since vanilla beedle's shop will cause an error since it looks for `Bug Net` when it should be looking for `Progressive Bug Net`, but the others should be changed as well for consistency: `Beedle - 50 Rupee Item`, `Faron Woods - Slingshot`, and `Sandship - Bow`.

```yaml
Beedle - 50 Rupee Item:
  original item: Progressive Bug Net
  type: skyloft, beedle, cheap
  Paths:
    - ShpSmpl/25
```
```yaml
Faron Woods - Slingshot:
  original item: Progressive Slingshot
  type: faron, long, combat #because of the Lopsa Bokoblin fight, this is a combat reward
  Paths:
    - event/200-Forest/595
    - oarc/F100/l0
  hint: sometimes
  text: the <r<Kikwi relic>> is
```
```yaml
Sandship - Bow:
  original item: Progressive Bow
  type: lanayru, dungeon
  Paths:
    - stage/D301/r15/l0/TBox/72
  hint: sometimes
  text: a <r<robot upon the Sandship's bow>> guards
```

**Comment by RaidWhipple** [comment]
Changed "Swift Beetle" to "Quick Beetle"
Changed `original item:` in `checks.yaml` to be progressive.

File: SS Rando Logic - Glitchless Requirements.yaml [hunk 1]
```diff
@@ -37,7 +37,19 @@ Hook Beetle:
   Progressive Beetle x2
 
 Quick Beetle:
-  Hook Beetle & Can Access Deep Woods & Can Activate Mines First Timeshift Stone & Can Defeat Bokoblins
+  (Hook Beetle & Emerald Tablet & Amber Tablet & Can Defeat Bokoblins & Option "gondo-upgrades" Disabled) | (Progressive Beetle x3 & Option "gondo-upgrades" Enabled)
```
Comment (lepelog) [inline SS Rando Logic - Glitchless Requirements.yaml:40]:
```suggestion
  (Hook Beetle & Emerald Tablet & Amber Tablet & Can Defeat Bokoblins & Option "gondo-upgrades" Disabled & Upgrade Hook Beetle Trick) | Progressive Beetle x3
```

There are only 2 beetles in the pool without the option enabled. In the future plandomizers might be able to alter the item pool, so even without the option getting 3 beetles could be possible.

File: SS Rando Logic - Glitchless Requirements.yaml [hunk 2]
```diff
@@ -37,7 +37,19 @@ Hook Beetle:
...
+
+Tough Beetle:
+  (Progressive Beetle x4 & Option "gondo-upgrades" Enabled)
```
Comment (lepelog) [inline SS Rando Logic - Glitchless Requirements.yaml:43]:
```suggestion
  Progressive Beetle x4
```

**Review by lepelog — CHANGES_REQUESTED** [review]

File: SS Rando Logic - Glitched Requirements.yaml [hunk 3]
```diff
@@ -37,7 +37,10 @@ Hook Beetle:
   Progressive Beetle x2
 
 Quick Beetle:
-  Hook Beetle & Can Access Deep Woods & Can Activate Mines First Timeshift Stone & Can Defeat Bokoblins
+  Progressive Beetle x3
```
Comment (lepelog) [inline SS Rando Logic - Glitched Requirements.yaml:40]:
```suggestion
  (Hook Beetle & Can Access Deep Woods & Can Activate Mines First Timeshift Stone & Can Defeat Bokoblins) | Progressive Beetle x3
```

File: SS Rando Logic - Glitchless Requirements.yaml [hunk 4]
```diff
@@ -37,7 +37,19 @@ Hook Beetle:
   Progressive Beetle x2
 
 Quick Beetle:
-  Hook Beetle & Can Access Deep Woods & Can Activate Mines First Timeshift Stone & Can Defeat Bokoblins
+  (Hook Beetle & Emerald Tablet & Amber Tablet & Can Defeat Bokoblins & Option "gondo-upgrades" Disabled) | Progressive Beetle x3
```
Comment (lepelog) [inline SS Rando Logic - Glitchless Requirements.yaml:40]:
```suggestion
  (Hook Beetle & Emerald Tablet & Amber Tablet & Can Defeat Bokoblins & Option "gondo-upgrades" Disabled & Upgrade Hook Beetle Trick) | Progressive Beetle x3
```

**Review by lepelog — CHANGES_REQUESTED** [review]

File: SS Rando Logic - Glitchless Requirements.yaml (hunk 4 above)
Comment (RaidWhipple) [reply SS Rando Logic - Glitchless Requirements.yaml:40]: In this situation, what would happen if Rupeesanity is fully enabled but the upgrade hook beetle trick is disabled?  There would be enabled checks that you can't logically reach.  If that's alright I can commit this but that's why I removed the trick.

Comment (lepelog) [reply SS Rando Logic - Glitchless Requirements.yaml:40]: oh that's right, we might need to rethink about that trick then and the interaction with rupeesanity, but for now it's fine I guess

**Review by lepelog — APPROVED** [review]

**Pull request merged**